nixos/modules/services/web-apps/sftpgo.nix

```diff
@@ -0,0 +1,370 @@
+                  description = mdDoc "Port of the SMTP Server.";
+                };
+
+                encryption = mkOption {
+                  type = types.int;
+                  default = 1;
+                  description = mdDoc ''
+                    Encryption scheme:
+                    0: No encryption
+                    1: TLS
+                    2: STARTTLS
+                  '';
+                };
+
+                auth_type = mkOption {
+                  type = types.int;
+                  default = 0;
+                  description = mdDoc ''
+                    0: Plain
+                    1: Login
+                    2: CRAM-MD5
+                  '';
+                };
+
+                user = mkOption {
+                  type = types.str;
+                  default = "sftpgo";
+                  description = mdDoc "SMTP username.";
+                };
+
+                from = mkOption {
+                  type = types.str;
+                  default = "SFTPGo <sftpgo@example.com>";
+                  description = mdDoc ''
+                    From address.
+                  '';
+                };
+              };
+            };
+          };
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.sftpgo.settings = (mapAttrs (name: mkDefault) {
+      ftpd.bindings = [{ port = 0; }];
+      httpd.bindings = [{ port = 0; }];
+      sftpd.bindings = [{ port = 0; }];
+      webdavd.bindings = [{ port = 0; }];
+      httpd.openapi_path = "${cfg.package}/openapi";
+      httpd.templates_path = "${cfg.package}/templates";
+      httpd.static_files_path = "${cfg.package}/static";
+    });
+
+    users = optionalAttrs (cfg.user == defaultUser) {
+      users = {
+        ${defaultUser} = {
+          description = "SFTPGo system user";
+          isSystemUser = true;
+          group = defaultUser;
+          home = cfg.dataDir;
+          createHome = true;
+        };
+      };
+
+      groups = {
+        ${defaultUser} = {
+          members = [ defaultUser ];
+        };
+      };
+    };
+
+    systemd.tmpfiles.rules = [
+      "d ${cfg.dataDir} 0750 ${cfg.user} ${cfg.group} -"
+    ];
+
+    systemd.services.sftpgo = {
+      description = "SFTPGo daemon";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      environment =
+        recursiveUpdate (mapAttrs (name: mkDefault) {
+          SFTPGO_CONFIG_FILE = configFile;
+          SFTPGO_LOG_FILE_PATH = ""; # log to journal
+        }) (optionalAttrs (cfg.loadDataFile != null) {
+          SFTPGO_LOADDATA_FROM = cfg.loadDataFile;
+        });
+
+      serviceConfig = {
+        Type = "simple";
+        User = cfg.user;
+        Group = cfg.group;
+        StateDirectory = "sftpgo";
+        ReadWritePaths = cfg.dataDir;
+        LimitNOFILE = 8192; # taken from upstream
+        KillMode = "mixed";
+        ExecStart = "${cfg.package}/bin/sftpgo serve ${escapeShellArgs cfg.extraArgs}";
```

Comment:
`Exec*` aren't actually shell, `utils.escapeSystemdExecArgs` is better suited

Reply:
Good to know, thanks!